### DISCUSSION TITLE
Custom element for initial focus when opening a popover

### DISCUSSION THREAD

**Discussion by arequejo** [radix-ui/primitives#915] [Help] [answered]
As of right now I have a popover where its first focusable element is not the close button (design team decisions) and the first focusable item is _not_ the least destructive action, in these scenarios we might want the focus on a different element as seen [here](https://www.w3.org/TR/wai-aria-practices/#keyboard-interaction-7). Is there a way to achieve this?

Thank you for the awesome library and all the hard work!

**Comment by benoitgrelard** [comment, marked as answer]
Hi @arequejo, yes you can use the `onOpenAutoFocus` prop and prevent our default behaviour (`event.preventDefault`) and run your own focus then instead.

![image](https://user-images.githubusercontent.com/1539897/137173373-211318cf-05ba-4d4e-abd6-03ff96455b25.png)

**Reply by arequejo** [reply]
My bad, thank you so much!
